pulsar/consumer.go

```diff
@@ -248,4 +257,13 @@ type Consumer interface {
 
 	// Name returns the name of consumer.
 	Name() string
+
+	// lastDequeuedMsg used for setting last dequeued msg id by internal partition consumers
```

Comment:
I don't think we should be adding private methods to the public interface. I think these should be removed. Also, it seems these are only relevant to the reader implementation so I think these helper functions should live within the reader files.